flyway-core/src/main/java/org/flywaydb/core/internal/util/scanner/classpath/ClassPathScanner.java

```diff
@@ -308,11 +308,16 @@ private ClassPathLocationScanner createLocationScanner(String protocol) {
         }
 
         if ("jar".equals(protocol)
-                || "war".equals(protocol)
+                || "war".equals(protocol) //Tomcat
                 || "zip".equals(protocol) //WebLogic
                 || "wsjar".equals(protocol) //WebSphere
                 ) {
-            JarFileClassPathLocationScanner locationScanner = new JarFileClassPathLocationScanner();
+            JarFileClassPathLocationScanner locationScanner;
+            if ("war".equals(protocol)) {
+                locationScanner = new JarFileClassPathLocationScanner("*/");
```

Comment:
I don't think so, as older tomcat version will return URLs like jar:xxx!/ for war files, and new tomcats will return URLs like war:xxx*/. but somebody should test this...